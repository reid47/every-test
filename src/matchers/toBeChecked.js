export default function toBeChecked(received) {
  const pass = !!received.checked;
  const message = pass ? `Expected element not to be checked.` : `Expected element to be checked.`;
  return { pass, message };
}
