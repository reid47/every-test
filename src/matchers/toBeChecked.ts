export default function toBeChecked(received: HTMLElement) {
  const pass = received instanceof HTMLInputElement && received.checked;
  const message = pass ? `Expected element not to be checked.` : `Expected element to be checked.`;
  return { pass, message };
}
