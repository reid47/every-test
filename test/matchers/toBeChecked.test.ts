import toBeChecked from '../../src/matchers/toBeChecked';

describe('toBeChecked', () => {
  test('passing', () => {
    const input = document.createElement('input');
    input.checked = true;

    const result = toBeChecked(input);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const input = document.createElement('input');
    input.checked = false;

    const result = toBeChecked(input);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
