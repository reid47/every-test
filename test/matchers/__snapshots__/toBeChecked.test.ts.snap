// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeChecked failing 1`] = `"Expected element to be checked."`;

exports[`toBeChecked passing 1`] = `"Expected element not to be checked."`;
